src/main/java/com/zendesk/maxwell/schema/SchemaStore.java

```diff
@@ -0,0 +1,197 @@
+package com.zendesk.maxwell.schema;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.zendesk.maxwell.BinlogPosition;
+import com.zendesk.maxwell.schema.columndef.ColumnDef;
+import com.zendesk.maxwell.schema.ddl.SchemaSyncError;
+
+public class SchemaStore {
+	private final Connection connection;
+	private Schema schema;
+	private BinlogPosition position;
+	static final Logger LOGGER = LoggerFactory.getLogger(SchemaStore.class);
+	private final PreparedStatement schemaInsert, databaseInsert, tableInsert,
+			columnInsert;
+
+	public SchemaStore(Connection connection) throws SQLException {
+		this.connection = connection;
+		this.schemaInsert = connection
+				.prepareStatement(
+						"INSERT INTO `maxwell`.`schemas` SET binlog_file = ?, binlog_position = ?, server_id = ?",
+						Statement.RETURN_GENERATED_KEYS);
+		this.databaseInsert = connection
+				.prepareStatement(
+						"INSERT INTO `maxwell`.`databases` SET schema_id = ?, name = ?, encoding=?",
+						Statement.RETURN_GENERATED_KEYS);
+		this.tableInsert = connection
+				.prepareStatement(
+						"INSERT INTO `maxwell`.`tables` SET schema_id = ?, database_id = ?, name = ?, encoding=?",
+						Statement.RETURN_GENERATED_KEYS);
+		this.columnInsert = connection
+				.prepareStatement(
+						"INSERT INTO `maxwell`.`columns` SET schema_id = ?, table_id = ?, name = ?, encoding=?, coltype=?, is_signed=?",
+						Statement.RETURN_GENERATED_KEYS);
+	}
+
+	public SchemaStore(Connection connection, Schema schema,
+			BinlogPosition position) throws SQLException {
+		this(connection);
+		this.schema = schema;
+		this.position = position;
+	}
+
+	private static Integer executeInsert(PreparedStatement preparedStatement,
+			Object... values) throws SQLException {
+		for (int i = 0; i < values.length; i++) {
+			preparedStatement.setObject(i + 1, values[i]);
+		}
+		preparedStatement.executeUpdate();
+
+		ResultSet rs = preparedStatement.getGeneratedKeys();
+
+		if (rs.next()) {
+			return rs.getInt(1);
+		} else
+			return null;
+	}
+
+	public void save() throws SQLException, IOException {
+		createStoreDatabase();
+
+		if (this.schema == null)
+			throw new RuntimeException("Uninitialized schema!");
+
+		Integer schemaId = executeInsert(schemaInsert, position.getFile(),
+				position.getOffset(), 1);
+
+		for (Database d : schema.getDatabases()) {
```

Comment:
Also, absolutely no error checks. Is it ok to fail catastrophically here if the save crashes?